config/webpack/plugin/GeneratePlugin.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Generate the html file
+ * @param {*} object
+ * filename - name for the generated html file
+ * lang - language for the page
+ * siteConfig - config for the page
+ * siteEntries - all entries
+ * entry - this entry
+ */
+const addHtmlPlugin = ({
+  filename, siteConfig, siteEntries, entry,
+}) => {
+  const otherSiteEntries = siteEntries.filter(indexEntry => indexEntry !== entry);
+  return new HtmlWebpackPlugin({
+    title: siteConfig.appConfig.title,
+    filename,
+    template: path.join(__dirname, '..', '..', '..', 'lib', 'index.html'),
+    rootElementId: 'root',
+    favicon: siteConfig.appConfig.favicon,
+    headHtml: [getNewRelicJS()].concat(siteConfig.appConfig.headHtml),
+    headChunks: ['rewriteHistory'],
+    excludeChunks: ['redirect', ...otherSiteEntries],
+    inject: false, // This turns off auto injection. We handle this ourselves in the template.
+  });
+};
+
+/**
+ * Generate the prefixed entry name
+ * @param {*} site
+ */
+const prefixEntry = site => (site.prefix ? `${site.prefix}/${'index'}` : 'index');
+
+/**
+ * Generate app title based on the app config.
+ * @param {*} site
+ */
+const appTitle = (site) => {
+  const { headline, title, subline } = site.siteConfig.appConfig;
+
+  return [headline, title, subline].filter(item => item).join(' - ');
+};
+
+/**
+ * Generate the config files needed for the sites passed in.
+ */
+class GeneratePlugin {
+  constructor({
+    sites, basename = '',
+  } = {}) {
+    this.entries = [];
+    this.apps = [];
+    this.sites = sites.map((site) => {
+      const entry = prefixEntry(site);
+      this.entries.push(entry);
+
+      let filename = 'index.html';
+      let siteBasename = basename;
+
+      // update filename and basename with the prefix for the site
+      if (site.prefix) {
+        filename = `${site.prefix}/index.html`;
+        siteBasename = `${basename}/${site.prefix}`;
+      }
+
+      // list of all apps to switch to
+      this.apps.push({
+        path: site.prefix,
+        url: siteBasename || '/',
+        title: appTitle(site),
+      });
+      return ({
+        ...site,
+        // name of webpack entry to be added
+        entry,
+        // name of html file to be generated
+        filename,
+        // basename for react router
+        basename: siteBasename,
+      });
+    });
+  }
+
+  apply(compiler) {
+    this.sites.forEach((site) => {
+      const {
+        siteConfig, prefix, basename, filename, entry, indexPath,
+      } = site;
+
+      // Add the entry to options, entry should already be valued.
+      compiler.options.entry[entry] = indexPath;
+
+      // Generate the files need to spin up the site.
+      generateAppConfig({
+        siteConfig,
+        mode: compiler.options.mode,
+        prefix,
+        apps: this.apps.filter(app => app.path !== prefix),
+        basename,
+      });
+
+      // generate index html files
+      addHtmlPlugin({
+        filename,
+        siteConfig,
+        siteEntries: this.entries,
```

Comment:
is `this.entries` scoped to this site?

Reply:
no, entries is all site entries to be added, and is filtered in the add html plugin function.

Reply:
Ahh so we can ignore the other entries chunks. 👍 nice

Reply:
I could prefilter in the constructor, but it would require an additional loop anyway, once to gather the entries and once to filter out the sites's entry. I could move the filtering out of the addhtmlplugin function if that would be more clear.

Reply:
We could probably just update the parameter comment to expain the intent:
https://github.com/cerner/terra-dev-site/pull/192/files/805a419593fde70d95148fad2c52cc334ef84c5b#diff-1c02ed9bfa3bd025833b739bf2c63ee6R14
maybe something like
siteEntries - all entries **to filter the include chunk assets**